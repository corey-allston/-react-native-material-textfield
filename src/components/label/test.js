import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import Label from '.';

/* eslint-env jest */

const string = 'test';
const props = {
  fontSize: 16,
  activeFontSize: 12,
  tintColor: 'blue',
  baseColor: 'black',
  errorColor: 'red',
  animationDuration: 225,
};

it('renders label', () => {
  let field = renderer
    .create(<Label {...props}>{string}</Label>)
    .toJSON();

  let text = field.children[0];

  expect(text.type).toBe('Text');
  expect(text.children).toEqual([string]);

  expect(field)
    .toMatchSnapshot();
});
