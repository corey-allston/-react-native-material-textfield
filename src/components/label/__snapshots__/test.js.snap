// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders label 1`] = `
<View
  style={
    Object {
      "position": "absolute",
      "top": 36,
    }
  }
>
  <Text
    accessible={true}
    active={false}
    allowFontScaling={true}
    animationDuration={225}
    disabled={false}
    ellipsizeMode="tail"
    errored={false}
    focused={false}
    numberOfLines={1}
    style={
      Object {
        "color": "rgba(0, 0, 0, 1)",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;
